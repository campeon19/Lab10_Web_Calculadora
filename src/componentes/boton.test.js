import React from 'react';
import ReactDOM from 'react-dom';
import { screen, render, fireEvent, cleanup } from '@testing-library/react';
import Boton from './boton';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

afterEach(cleanup);
it('renders without crashing', ()=>{
    const div = document.createElement('div');
    ReactDOM.render(<Boton />, div)
});

it('renders button correctly', ()=>{
    const { getByTestId } = render(<Boton name = '+'></Boton>)
    expect(getByTestId('boton')).toHaveTextContent('+')
});

it('renders button correctly', ()=>{
    const { getByTestId } = render(<Boton name = '+/-'></Boton>)
    expect(getByTestId('boton')).toHaveTextContent('+/-')
});

it('renders button correctly', ()=>{
    const { getByTestId } = render(<Boton name = '='></Boton>)
    expect(getByTestId('boton')).toHaveTextContent('=')
});

it('matches snapshot', () =>{
    const tree = renderer.create(<Boton name='-'/>).toJSON();
    expect(tree).toMatchSnapshot();
});