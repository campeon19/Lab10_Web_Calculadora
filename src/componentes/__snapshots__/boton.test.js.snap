// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="component-button"
>
  <button
    data-testid="boton"
    onClick={[Function]}
  >
    -
  </button>
</div>
`;
